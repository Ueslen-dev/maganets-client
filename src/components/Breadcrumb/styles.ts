import styled, { css } from 'styled-components';

export const Wrapper = styled.ul`
  display: flex;

  li:last-child {
    span {
      display: none;
    }
  }

  ${({ theme }) => css`
    gap: ${theme.spacings.xxsmall};
  `}
`;

export const Routes = styled.li`
  list-style: none;

  ${({ theme }) => css`
    a {
      font-weight: ${theme.font.bold};
      font-size: ${theme.font.sizes.large};
      color: ${theme.colors.black};
      text-decoration: none;

      &:hover {
        color: ${theme.colors.primary};
        transition: 0.5s all;
      }
    }
    span {
      margin-left: ${theme.spacings.xxsmall};
      & + {
        display: none;
      }
    }
  `}
`;
