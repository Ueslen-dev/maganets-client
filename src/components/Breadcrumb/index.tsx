import Link from 'next/link';

import * as S from './styles';

type Props = {
  routes: { id: number; to: string; name: string }[];
};
const Breadcrumb = ({ routes }: Props) => {
  const arrow = '>';
  return (
    <S.Wrapper>
      {routes?.map((route) => (
        <S.Routes key={route.id} data-testid="breadcrumb">
          <Link href={route.to}>
            <a>{route.name}</a>
          </Link>
          <span>{arrow}</span>
        </S.Routes>
      ))}
    </S.Wrapper>
  );
};
export default Breadcrumb;
