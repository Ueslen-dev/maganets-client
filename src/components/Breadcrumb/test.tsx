import { screen, render } from '@testing-library/react';

import Breadcrumb from '.';

describe('<Breadcrumb/>', () => {
  it('should render the heading', () => {
    const { container } = render(<Breadcrumb />);
    expect(
      screen.getByRole('heading', { name: /Breadcrumb/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

});
