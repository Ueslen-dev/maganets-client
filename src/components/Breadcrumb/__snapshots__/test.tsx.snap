// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb/> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Breadcrumb
  </h1>
</main>
`;
